common/courses/states.ts

```diff
@@ -129,15 +131,41 @@ export async function editSubcourse(subcourse: Subcourse, update: Partial<Subcou
     if (!can.allowed) {
         throw new Error(`Cannot edit Subcourse(${subcourse.id}) reason: ${can.reason}`);
     }
+    const participantCount = await prisma.subcourse_participants_pupil.count({ where: { subcourseId: subcourse.id } });
 
     const isMaxParticipantsChanged: boolean = Boolean(update.maxParticipants);
+    const isGroupChatTypeChanged: boolean = Boolean(update.groupChatType);
 
     if (isMaxParticipantsChanged) {
-        const participantCount = await prisma.subcourse_participants_pupil.count({ where: { subcourseId: subcourse.id } });
         if (update.maxParticipants < participantCount) {
             throw new PrerequisiteError(`Decreasing the number of max participants below the current number of participants is not allowed`);
         }
     }
+    if (isGroupChatTypeChanged) {
+        //* if the subcourse has already an conversation, the conversation has to be updated
+        if (subcourse.conversationId !== null) {
+            if (update.groupChatType === ChatType.NORMAL) {
+                const conversationToBeUpdated: { id: string } & ConversationInfos = {
+                    id: subcourse.conversationId,
+                    custom: {
+                        type: 'course',
+                    },
+                };
+
+                await markConversationAsWriteable(subcourse.conversationId);
+                await updateConversation(conversationToBeUpdated);
+            } else if (update.groupChatType === ChatType.ANNOUNCEMENT) {
+                const conversationToBeUpdated: { id: string } & ConversationInfos = {
+                    id: subcourse.conversationId,
+                    custom: {
+                        type: 'announcement',
+                    },
+                };
+                await markConversationAsReadOnly(subcourse.conversationId);
```

Comment:
Just a question: in some cases we want to mark the whole conversation as readonly and in this case we want to mark only the participants (not the instructors) as readonly, is this handled by this function? Have you thought of this?